src/libraries/System.Text.Json/src/System/Text/Json/Serialization/WriteStack.cs

```diff
@@ -180,6 +208,156 @@ public void Pop(bool success)
             }
         }
 
+        // asynchronously await any pending stacks that resumable converters depend on.
+        public async ValueTask AwaitPendingTask()
+        {
+            Debug.Assert(PendingTask != null);
+
+            if (!PendingTask.IsCompleted)
+            {
+                // wrap with Task.WhenAny to avoid surfacing any exceptions here
+                await Task.WhenAny(PendingTask).ConfigureAwait(false);
+            }
+
+            // Do not clear the `PendingTask` field here since the result
+            // will need to be consumed by a resumable converter.
+        }
```

Comment:
```suggestion
        public Task AwaitPendingTask()
        {
            Debug.Assert(PendingTask != null);
            return PendingTask.IsCompleted ?
                PendingTask :
                Task.WhenAny(PendingTask); // wrap with Task.WhenAny to avoid surfacing any exceptions here

            // Do not clear the `PendingTask` field since the result
            // will need to be consumed by a resumable converter.
        }
```

Reply:
If `PendingTask` is completed with an exception, then it would result in the exception being surfaced here. This is something I want to explicitly avoid, since it needs to be surfaced in the context of the resumable converter.

Reply:
Hmm.  This means that TaskScheduler.UnobservedTaskException might start firing with ignored exceptions?  Or are those exceptions going to be propagated somewhere else?

Reply:
The idea is to pass the completed task back to the resumable converter (which cannot be async) and surface any exceptions there via `Task.Result`.

Reply:
https://github.com/dotnet/runtime/pull/50778/files#diff-dbb426b9ad2dff2e1cac3fd92f39ec28124ecc7ff0c151959be7fb519623e55dR47

Reply:
Ok, cool.